Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -297,6 +302,11 @@ public partial class Scheduler : IPipScheduler, IPipExecutionEnvironment, IFileC
         private int m_maxUnresponsivenessFactor = 0;
         private DateTime m_statusLastCollected = DateTime.MaxValue;
 
+        private ConcurrentDictionary<string, long> m_aggregatedPipPropertiesCount = new ConcurrentDictionary<string, long>();
+        private ConcurrentDictionary<string, HashSet<string>> m_addgregatepipsPerPipProperty = new ConcurrentDictionary<string, HashSet<string>>();
+        private ConcurrentBigSet<string> m_pipsSucceedingAfterUserRetry = new ConcurrentBigSet<string>();
+        private ConcurrentBigSet<string> m_pipsFailingAfterLastUserRetry = new ConcurrentBigSet<string>();
```

Comment:
Can you please create a separate class containing all of these `ConcurrentDictionary` and `ConcurrentBugSet`?